pancancer_evaluation/prediction/classification.py

```diff
@@ -0,0 +1,355 @@
+    metric_df = pd.DataFrame(metrics_, columns=metric_cols)
+    gene_auc_df = pd.concat([train_roc_df, test_roc_df, cv_roc_df])
+    gene_aupr_df = pd.concat([train_pr_df, test_pr_df, cv_pr_df])
+
+    return metric_df, gene_auc_df, gene_aupr_df
+
+
+def get_metrics_cc(y_train_df, y_test_df, y_cv_df, y_pred_train,
+                   y_pred_test, train_identifier, test_identifier,
+                   signal, seed, train_pancancer=False):
+
+    # get classification metric values
+    y_train_results = get_threshold_metrics(
+        y_train_df.status, y_pred_train, drop=False
+    )
+    y_test_results = get_threshold_metrics(
+        y_test_df.status, y_pred_test, drop=False
+    )
+    y_cv_results = get_threshold_metrics(
+        y_train_df.status, y_cv_df, drop=False
+    )
+
+    # summarize all results in dataframes
+    if train_pancancer:
+        metric_cols = [
+            "auroc",
+            "aupr",
+            "train_gene",
+            "test_identifier",
+            "signal",
+            "seed",
+            "data_type"
+        ]
+    else:
+        metric_cols = [
+            "auroc",
+            "aupr",
+            "train_identifier",
+            "test_identifier",
+            "signal",
+            "seed",
+            "data_type"
+        ]
+    train_metrics_, train_roc_df, train_pr_df = summarize_results_cc(
+        y_train_results, train_identifier, test_identifier,
+        signal, seed, "train"
+    )
+    test_metrics_, test_roc_df, test_pr_df = summarize_results_cc(
+        y_test_results, train_identifier, test_identifier,
+        signal, seed, "test"
+    )
+    cv_metrics_, cv_roc_df, cv_pr_df = summarize_results_cc(
+        y_cv_results, train_identifier, test_identifier,
+        signal, seed, "cv"
+    )
+
+    # compile summary metrics
+    metrics_ = [train_metrics_, test_metrics_, cv_metrics_]
+    metric_df = pd.DataFrame(metrics_, columns=metric_cols)
+    gene_auc_df = pd.concat([train_roc_df, test_roc_df, cv_roc_df])
+    gene_aupr_df = pd.concat([train_pr_df, test_pr_df, cv_pr_df])
+
+    return metric_df, gene_auc_df, gene_aupr_df
+
+
+def summarize_results(results, gene, holdout_cancer_type, signal, seed,
+                      data_type, fold_no):
+    """
+    Given an input results file, summarize and output all pertinent files
+
+    Arguments
+    ---------
+    results: a results object output from `get_threshold_metrics`
+    gene: the gene being predicted
+    holdout_cancer_type: the cancer type being used as holdout data
+    signal: the signal of interest
+    seed: the seed used to compress the data
+    data_type: the type of data (either training, testing, or cv)
+    fold_no: the fold number for the external cross-validation loop
+    """
+    results_append_list = [
+        gene,
+        holdout_cancer_type,
+        signal,
+        seed,
+        data_type,
+        fold_no,
+    ]
+
+    metrics_out_ = [results["auroc"], results["aupr"]] + results_append_list
```

Comment:
Is there a reason for the trailing underscore here? Is there a `metrics_out` variable that you're avoiding a conflict with?

Reply:
No reason that I know of. I copied this part of the code from one of Greg's repos a long time ago, so I'm not sure exactly what he was going for. It looks like the trailing underscore in sklearn code is sometimes used for [variables that are only defined after the model is fit](https://stackoverflow.com/a/66205119), which I guess would apply here, but this function really only makes sense to call after the model is fit anyway so I don't think the trailing underscore adds much. I'll remove it here!